src/main/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtils.java

```diff
@@ -173,4 +203,15 @@ public static boolean fileExists(String path, PipelineOptions popts) {
         }
         return true;
     }
+
+    /**
+     * Create a new GCS options with a given api key.
+     * @param apiKey a valid GCS api key
+     */
+    public static GCSOptions getAuthenticatedGCSOptions(String apiKey) {
```

Comment:
This was just a convenience method added because spark currently only supports the apiKey, but your point about not introducing code paths that understand one but not the other is fair -- I've removed this.